// angular-calendar-scheduler/esm2022/modules/scheduler/utils/calendar-scheduler-utils.provider.mjs
import { Inject, Injectable } from '@angular/core';
import { getSchedulerViewHourGrid, getSchedulerViewDays, getSchedulerView } from './calendar-scheduler-utils';
import { MOMENT } from 'angular-calendar';
import * as i0 from "@angular/core";
import * as i1 from "angular-calendar";
export class CalendarSchedulerUtils {
    dateAdapter;
    moment;
    constructor(dateAdapter, moment) {
        this.dateAdapter = dateAdapter;
        this.moment = moment;
    }
    getSchedulerViewHourGrid(args) {
        return getSchedulerViewHourGrid(this.dateAdapter, args);
    }
    getSchedulerViewDays(args) {
        return getSchedulerViewDays(this.dateAdapter, args);
    }
    getSchedulerView(args) {
        return getSchedulerView(this.dateAdapter, this.moment, args);
    }
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.7", ngImport: i0, type: CalendarSchedulerUtils, deps: [{ token: i1.DateAdapter }, { token: MOMENT }], target: i0.ɵɵFactoryTarget.Injectable });
    static ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.2.7", ngImport: i0, type: CalendarSchedulerUtils });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.7", ngImport: i0, type: CalendarSchedulerUtils, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.DateAdapter }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [MOMENT]
                }] }]; } });
