// angular-calendar-scheduler/esm2022/modules/scheduler/calendar-scheduler-event-content.component.mjs
import { Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export class CalendarSchedulerEventContentComponent {
    hostElement;
    event;
    eventContainer;
    maxHeight;
    constructor(hostElement) {
        this.hostElement = hostElement;
    }
    ngAfterViewInit() {
        setTimeout(() => { this.maxHeight = Math.max(30, this.eventContainer.clientHeight - 70); }, 0);
    }
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.7", ngImport: i0, type: CalendarSchedulerEventContentComponent, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.7", type: CalendarSchedulerEventContentComponent, selector: "calendar-scheduler-event-content", inputs: { event: "event", eventContainer: "eventContainer" }, host: { classAttribute: "cal-scheduler-event-content-container" }, ngImport: i0, template: `
        <div *ngIf="event.content"
            class="cal-scheduler-event-content"
            [style.max-height.px]="maxHeight"
            [style.white-space]="maxHeight && maxHeight > 30 ? 'normal' : 'nowrap'"
            [innerHTML]="event.content">
        </div>
    `, isInline: true, dependencies: [{ kind: "directive", type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.7", ngImport: i0, type: CalendarSchedulerEventContentComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'calendar-scheduler-event-content',
                    template: `
        <div *ngIf="event.content"
            class="cal-scheduler-event-content"
            [style.max-height.px]="maxHeight"
            [style.white-space]="maxHeight && maxHeight > 30 ? 'normal' : 'nowrap'"
            [innerHTML]="event.content">
        </div>
    `,
                    host: {
                        'class': 'cal-scheduler-event-content-container'
                    }
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { event: [{
                type: Input
            }], eventContainer: [{
                type: Input
            }] } });
