// angular-calendar-scheduler/esm2022/modules/scheduler/scheduler-config.mjs
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * Auth configuration.
 */
export class SchedulerConfig {
    locale = 'en';
    headerDateFormat = 'daysRange';
    logEnabled = false;
    constructor(config = {}) {
        function use(source, defaultValue) {
            return config && source !== undefined ? source : defaultValue;
        }
        this.locale = use(config.locale, this.locale);
        this.headerDateFormat = use(config.headerDateFormat, this.headerDateFormat);
        this.logEnabled = use(config.logEnabled, this.logEnabled);
    }
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.7", ngImport: i0, type: SchedulerConfig, deps: [{ token: SchedulerConfig }], target: i0.ɵɵFactoryTarget.Injectable });
    static ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.2.7", ngImport: i0, type: SchedulerConfig });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.7", ngImport: i0, type: SchedulerConfig, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: SchedulerConfig }]; } });
