// angular-calendar-scheduler/esm2022/modules/scheduler/pipes/calendar-scheduler-date.pipe.mjs
import { Pipe, LOCALE_ID, Inject } from '@angular/core';
import { CalendarDateFormatter } from 'angular-calendar';
import * as i0 from "@angular/core";
import * as i1 from "angular-calendar";
export class CalendarSchedulerDatePipe {
    dateAdapter;
    dateFormatter;
    locale;
    constructor(dateAdapter, dateFormatter, locale) {
        this.dateAdapter = dateAdapter;
        this.dateFormatter = dateFormatter;
        this.locale = locale;
    }
    transform(date, method, locale = this.locale, weekStartsOn = 0, excludeDays = [], daysInWeek, startsWithToday) {
        if (typeof this.dateFormatter[method] === 'undefined') {
            const allowedMethods = Object.getOwnPropertyNames(Object.getPrototypeOf(CalendarDateFormatter.prototype)).filter(iMethod => iMethod !== 'constructor');
            throw new Error(`${method} is not a valid date formatter. Can only be one of ${allowedMethods.join(', ')}`);
        }
        return this.dateFormatter[method]({
            dateAdapter: this.dateAdapter,
            date,
            locale,
            weekStartsOn,
            excludeDays,
            daysInWeek,
            startsWithToday
        });
    }
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.7", ngImport: i0, type: CalendarSchedulerDatePipe, deps: [{ token: i1.DateAdapter }, { token: i1.CalendarDateFormatter }, { token: LOCALE_ID }], target: i0.ɵɵFactoryTarget.Pipe });
    static ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "16.2.7", ngImport: i0, type: CalendarSchedulerDatePipe, name: "calendarSchedulerDate" });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.7", ngImport: i0, type: CalendarSchedulerDatePipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'calendarSchedulerDate'
                }]
        }], ctorParameters: function () { return [{ type: i1.DateAdapter }, { type: i1.CalendarDateFormatter }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [LOCALE_ID]
                }] }]; } });
